src/types.rs

```diff
@@ -19,7 +19,8 @@ pub type NewStreamContext = fn(context_id: u32, root_context_id: u32) -> Box<dyn
 pub type NewHttpContext = fn(context_id: u32, root_context_id: u32) -> Box<dyn HttpContext>;
 
 #[repr(u32)]
-#[derive(Debug)]
+#[derive(Copy, Clone, Eq, PartialEq, Hash, Debug)]
+#[non_exhaustive]
```

Comment:
This is unlikely to change, and we should definitely allow exhaustive `match` on `LogLevel`.

Reply:
Feel free to replace this PR or edit it in place.